WordPress/gutenberg
# [RNMobile] Ensure uploaded audio is always visible within Audio block

* Gutenberg Mobile: https://github.com/wordpress-mobile/gutenberg-mobile/pull/6340

## What?

Fixes https://github.com/WordPress/gutenberg/issues/54328, by ensuring the Audio block is always visible when media's been uploaded to it.

## Why?

The content of an Audio block with set media is currently invisible when a block-based theme's background colour contrasts with the device's light/dark light mode. For example, if the theme's background colour is white and dark mode is enabled on a device, the Audio block's content will be invisible.

## How?

To account for a block based theme's background colour, a custom hook, `useEditorColorScheme`, has been created. The hook works as follows:

- Accepts both light and dark styles as its parameters.
- If no block-based theme is detected, `getStylesFromColorScheme()` is used to apply the appropriate style based on whether light or dark mode is enabled on a device. This is how the styles were previously applied. 
- If a block-based theme, its background colour is passed to the `colord.isDark()` function to determine if its dark or not. Then, the result is used to determine the appropriate colour scheme to use.

## Testing Instructions

Follow the _preparation steps_ once for each mode (dark/light) and then proceed with the _testing steps_.

### Preparation steps

* Ensure the device is in either dark mode or light mode.
* Activate a theme with support for global styles, e.g. [Byrne](https://byrnedemo.wordpress.com/).
* In your site's /wp-admin area, navigate to the site editor (Appearance → Editor) and select a style corresponding to the opposite mode (dark background for light mode, and vice versa).
* In the app, navigate to the editor for any of the site's posts or pages.

### Testing steps

* Tap the + icon to open the block inserter and add an Audio block.
* Follow the prompts to upload an audio file to the block.
* Verify that you're able to see the uploaded file within the block, along with prompts to play it.

### Additional testing

* Ensure there are no regressions associated with non-block based themes i.e. dark mode should still retain dark styles and vice versa.
* Experiment with non-conventional dark/light background colours with block based themes i.e. colours other than white or black.
* Test other changes styles, such as the failed upload styles.

### Testing Instructions for Keyboard

N/A, solely a visual change.

## Screenshots or screencast 

<details>
 <summary>Dark mode + block based theme with light background ⤵️</summary>
 
| Before  | After |
| ------------- | ------------- |
| <img src="https://github.com/WordPress/gutenberg/assets/2998162/98277a16-61b3-4dc2-8176-e9ae3e0a7000" width="100%"> | <img src="https://github.com/WordPress/gutenberg/assets/2998162/780e3408-687d-484a-a623-5820a8acb352" width="100%"> |
</details>

<details>
 <summary>Light mode + block based theme with dark background ⤵️</summary>
 
| Before  | After |
| ------------- | ------------- |
| <img src="https://github.com/WordPress/gutenberg/assets/2998162/71cc205d-d647-4562-a643-9f0decb22415" width="100%"> | <img src="https://github.com/WordPress/gutenberg/assets/2998162/6d43f8e4-185c-4fd6-ad63-9a4a01907022" width="100%"> |
</details>

<details>
<summary>No changes when dark/light mode matches background ⤵️</summary>
 
| Dark mode + dark background  | Light mode + light background |
| ------------- | ------------- |
| <img src="https://github.com/WordPress/gutenberg/assets/2998162/6ca03a7c-0833-4ce5-8d4a-82563409104b" width="100%"> | <img src="https://github.com/WordPress/gutenberg/assets/2998162/86ef5e25-7c75-405a-9dfc-79ad45cb44d4" width="100%"> |
</details>

<details>
 <summary>No changes for non-block based themes ⤵️</summary>
 
| Dark mode  | Light mode |
| ------------- | ------------- |
| <img src="https://github.com/WordPress/gutenberg/assets/2998162/d310ef07-7610-45cd-b5d4-ada9cfadc364" width="100%"> | <img src="https://github.com/WordPress/gutenberg/assets/2998162/7808e206-7857-4327-a741-23aadf06eef2" width="100%"> |
</details>
